submissions/nadsatt/movie-seat-booking-js/index.js

```diff
@@ -0,0 +1,111 @@
+  const SEAT_PRICE = 18;
+  const RENDERED_ORDER_INFO_ITEMS = {};
+
+  SUBMIT_BUTTON.addEventListener('click', (e) => {
+    e.preventDefault();
+    SUBMIT_BUTTON.disabled = true;
+    fillTicket(FORM_INPUTS);
+    showTicketWrapper();
+  });
+  TICKET_WRAPPER.addEventListener('click', () => {
+    SUBMIT_BUTTON.disabled = false;
+    hideTicketWrapper();
+    resetTicket();
+  });
+  SEAT_CONTROL.addEventListener('input', updateOrderInfoList);
+
+  function fillTicket(inputs) {
+    const checkedInputs = getCheckedInputs(inputs);
+    const values = getValues(checkedInputs);
+
+    TICKET_VALUES.forEach((ticketValue) =>
+      ticketValue.insertAdjacentHTML(
+        'afterbegin',
+        values[ticketValue.id] || '-'
+      )
+    );
+  }
+
+  function getCheckedInputs(inputs) {
+    return Array.prototype.filter.call(inputs, (input) => input.checked);
+  }
+
+  function getValues(inputs) {
+    const values = {};
+
+    values.day = inputs.find((input) => input.name === 'day').value;
+    values.time = inputs.find((input) => input.name === 'time').value;
+    values.hall = inputs.find((input) => input.name === 'time').dataset.hall;
+    values.seat = inputs
+      .filter((input) => input.name === 'seat')
+      .map(({ value }) => {
+        const [row, seat] = value.split('_');
+        return `<li class="ticket__seat-value-item">Row
+                  <span class="ticket__row-value">${row}</span>/ Seat
+                  <span class="ticket__seat-value">${seat}<span>
+                </li>`;
+      })
+      .join('');
+
+    return values;
+  }
+
+  function showTicketWrapper() {
+    TICKET_WRAPPER.classList.add('ticket-wrapper--visible');
+  }
+
+  function hideTicketWrapper() {
+    TICKET_WRAPPER.classList.remove('ticket-wrapper--visible');
+  }
+
+  function resetTicket() {
+    TICKET_VALUES.forEach((ticketValue) => (ticketValue.textContent = ''));
+  }
+
+  function updateOrderInfoList({ srcElement, srcElement: { value } }) {
+    if (srcElement.checked) {
+      addOrderInfoItem(value);
+    } else {
+      removeOrderInfoItem(value);
+    }
+  }
+
+  function addOrderInfoItem(value) {
+    const infoItem = createOrderInfoItem(value);
+    ORDER_INFO_LIST.append(infoItem);
+    RENDERED_ORDER_INFO_ITEMS[value] = infoItem;
+    ORDER_TOTAL_PRICE.textContent = +ORDER_TOTAL_PRICE.textContent + SEAT_PRICE;
+  }
+
+  function createOrderInfoItem(value) {
+    const [row, seat] = value.split('_');
+
+    const infoItem = document.createElement('li');
+    infoItem.classList.add('order__info-item');
+    infoItem.innerHTML = `<div>
+        <span>Row</span>
+        <span class="order__row-value">${row}</span>
+        <span>/&nbsp;Seat&nbsp;</span>
+        <span class="order__seat-value">${seat}</span>
+       </div>
+       <span class="order__price-value">$18</span>`;
+
+    return infoItem;
+  }
+
+  function removeOrderInfoItem(value) {
+    RENDERED_ORDER_INFO_ITEMS[value].remove();
+    delete RENDERED_ORDER_INFO_ITEMS[value];
```

Comment:
Not a good choice for removing elements from array. [Read this article](https://love2dev.com/blog/javascript-remove-from-array/) and choose the corresponding approach to delete elements from the array.

Reply:
but `RENDERED_ORDER_INFO_ITEMS` is object  👀 .

Reply:
Ok), my bad. 
What code on line 107 doing?

Reply:
It removes html element (which represents selected seat info) from the page. 

P.s. I use object `RENDERED_ORDER_INFO_ITEMS` (as storage for rendered elements) just because taking link to html element by key is easier than searching for html element in DOM.

Pps: if it is not clear, could you please advice me how I can fix it or which approach could be used instead